sections/vagrant.zsh

```diff
@@ -27,7 +27,15 @@ spaceship_vagrant() {
   # Show Vagrant status only for Vagrant-specific folders
   [[ -f Vagrantfile || (-n $VAGRANT_VAGRANTFILE && -f ${VAGRANT_VAGRANTFILE}) ]] || return
 
-  local vagrant_status=$(vagrant status | sed -n 3p | awk {'print $2'})
+  # local vagrant_status=$(cat .vagrant/machines/*/virtualbox/id | xargs VboxManage showvminfo | grep State | awk {'print $2'})
+
+  local vagrant_status=$(cat ${HOME}/.vagrant.d/data/machine-index/index | python -c 'import sys, os, json;
```

Comment:
@salmanulfarzy @maximbaz what do you think?

Reply:
> I don't use vagrant, could someone paste a real example of ${HOME}/.vagrant.d/data/machine-index/index and highlight the piece that this code is trying to extract?

Sure. [here](https://gist.github.com/guilhermeleobas/423dccb5f09441921a8e4f883be1d3f3#file-index-L14) is the content of the index file on my computer.

What I do is basically get the `state` of the machine whose `vagrantfile_path` is the same as the `pwd`.

>Just for the sake of completeness, I would be interested to see if the same can be achieved using awk or jq. Maybe this solution is the best one, but I haven't seen the alternatives yet :)

Maybe if `jq` would be easier but that adds and extra dependency.

Reply:
I played a little bit, here are my notes:

1. `python` seems to also be a new dependency, although I admit it is more likely to already be installed.
2. `jq`'s solution is surprisingly small and readable, less chance to make a typo:

```
cat ~/test.json | jq --arg dir "$PWD" '.machines[] | select(.vagrantfile_path == $dir).state'
```
3. `jq`'s solution above performs approximately 9 times faster than `python`'s one in my rough experiments.

You guys decide what is worse, having larger and slower python code or asking people to install `jq`. My personal vote is for `jq`, but I'm not using vagrant, so I will not spend time defending it 🙂

Reply:
I also prefer `jq` solution and I chose python because it comes already installed in most OS. But if you guys decide to move forward with `jq` solution, I can change without any problems.

Regards

Reply:
Can we make `jq` the preferred method and `python` as fallback ?

Reply:
> I also prefer `jq` solution and I chose `python` because it comes already installed in most OS. 

> Can we make `jq` the preferred method and `python` as fallback ?

I think `jq` can benefit us a lot in many places, it is worth adding it as a dependency and using it ubiquitously. We can stop parsing JSON with `grep` (e.g. https://github.com/denysdovhan/spaceship-prompt/pull/441 😉), and we won't need any fallback solutions, here and in other places.

Reply:
Sorry, Accidentally removed my comment while reviewing. 

> we won't need any fallback solutions,

Not against using `jq` everywhere. But, We can't expect every user to have `jq` installed, So fallback method would benefit users don't have `jq`.

Already working on #441 with `jq`(with fallback to `grep`) :smile:

Reply:
> But, We can't expect every user to have jq installed

Why not? `jq` is available in many (most?) official package repositories on various distributions, it is as easy as adding a note in README and `scripts/install.sh` that `jq` is a dependency that has to be installed by users.

And in case `jq` is absent, making sure that this section renders something like "jq is not available" if file "$HOME/.vagrant.d/data/machine-index/index" exists, and renders nothing if the file doesn't exist.

Reply:
> it is as easy as adding a note in README and `scripts/install.sh` that jq is a dependency that has to be installed by users.

Not in favor of requiring tool not pre-bundled with operating systems as only measure to render sections in prompt. We could take advantage if `jq` is already installed and may be add note about using `jq` will have improved performance.

Reply:
FYI `python` is not pre-bundled with all operating systems too, for example [here's the list of pre-bundled packages for Arch Linux](https://www.archlinux.org/groups/x86_64/base).